.github/workflows/sycl-containers.yaml

```diff
@@ -93,6 +93,33 @@ jobs:
             ghcr.io/${{ github.repository }}/ubuntu2204_intel_drivers:latest
           build-args: |
             use_latest=false
+  # This job produces a Docker container with the released versions of Intel
+  # drivers, that are specificed in dependencies.json., but use dev igc instead
+  drivers_image_ubuntu2204_devigc:
+    if: github.repository == 'intel/llvm'
+    name: Intel Drivers Ubuntu 22.04 Docker image with dev IGC
+    runs-on: ubuntu-22.04
+    permissions:
+      packages: write
+    needs: base_image_ubuntu2204
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - name: Build and Push Container
+        uses: ./devops/actions/build_container
+        with:
+          push: ${{ github.event_name != 'pull_request' }}
+          file: ubuntu2204_intel_drivers
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          tags: |
+            ghcr.io/${{ github.repository }}/ubuntu2204_intel_drivers:devigc-${{ github.sha }}
+            ghcr.io/${{ github.repository }}/ubuntu2204_intel_drivers:devigc
+          build-args: |
+            use_latest=false
+            use_igc_dev=true
```

Comment:
Would it be possible to refactor this code to use a `matrix`? See https://github.com/intel/llvm/blob/sycl/.github/workflows/sycl-linux-matrix-e2e-on-nightly.yml for an example.

Reply:
Sure, can do the factor in a follow up PR.

Reply:
Can we do that in the reverse order? Refactor first, then add new code.

Reply:
Fair enough.